### ISSUE TITLE
prov/opx: compliation failed because of dependency on numa.h

### ISSUE THREAD

**Issue by wzamazon** [ofiwg/libfabric#7567]
**Describe the bug**
prov/opx include `numa.h` because did not add related configure change.

**To Reproduce**
`./configure`
then `make`

**Expected behavior**
If `--enable-opx` is NOT explicitly set, opx need to check whether numa.h is present and disable itself from being built.
if `--enable-opx` is set, opx also need to check whether `numa.h` is present, and fail the configuration if `numa.h` is not available

**Output**
```
[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2298)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2299)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2300)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2301)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2302)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2303)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2304)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2305)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2306)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2307)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2308)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2309)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2310)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2311)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2312)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2313)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyTestsMain/detail/NightlyTestsMain/178/pipeline#step-66-log-2314)CC       prov/opx/src/src_libfabric_la-fi_opx_domain.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_ep.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_eq.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_fabric.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_info.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_init.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_mr.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_msg.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_rma.lo

  CC       prov/opx/src/src_libfabric_la-fi_opx_hfi1.lo

prov/opx/src/fi_opx_hfi1.c:34:10: fatal error: numa.h: No such file or directory

 #include <numa.h>

          ^~~~~~~~

compilation terminated.

make[1]: *** [prov/opx/src/src_libfabric_la-fi_opx_hfi1.lo] Error 1

make[1]: *** Waiting for unfinished jobs....

make[1]: Leaving directory `/fsx/libfabric'

make: *** [all] Error 2
```

**Environment:**
N/A

**Additional context**
N/A

**Comment by j-xiong** [comment]
@mwheinz

**Comment by j-xiong** [comment]
Fixed by #7583. Not sure why it is not closed automatically with the "Fixes" phrase.
